zookeeper-client/zookeeper-client-c/src/cli.c

```diff
@@ -678,15 +678,15 @@ int main(int argc, char **argv) {
     }
     if (argc > 2) {
       if(strncmp("cmd:",argv[2],4)==0){
-        size_t cmdlen = strlen(argv[2]);
+        size_t cmdlen = strlen(argv[2]) - 4;
         if (cmdlen > sizeof(cmd)) {
           fprintf(stderr,
                   "Command length %zu exceeds max length of %zu\n",
                   cmdlen,
                   sizeof(cmd));
           return 2;
         }
-        strncpy(cmd, argv[2]+4, sizeof(cmd));
+        memcpy(cmd, argv[2]+4, cmdlen + 1);
```

Comment:
This looks broken to me given the conditional above. "if (cmdlen > sizeof(cmd))"